iroh-sync/src/ranger.rs

```diff
@@ -611,25 +570,100 @@ where
         }
 
         // Now we remove all entries that have our key as a prefix and are older than our entry.
-        let removed = self
-            .store
-            .remove_prefix_filtered(entry.key(), |value| entry.value() >= value)?;
+        let removed = self.remove_prefix_filtered(entry.key(), |value| entry.value() >= value)?;
 
         // Insert our new entry.
-        self.store.put(entry)?;
+        self.kv_put(entry)?;
         Ok(InsertOutcome::Inserted { removed })
     }
+}
 
-    /// List all existing key value pairs.
-    // currently unused outside of tests
-    #[cfg(test)]
-    pub fn all(&self) -> Result<impl Iterator<Item = Result<E, S::Error>> + '_, S::Error> {
-        self.store.all()
+impl<E: RangeEntry, S: Store<E>> Store<E> for &mut S {
```

Comment:
It is a bit unforunate that rust does not provide a way to auto-impl this, but oh well, so be it...